### PR TITLE
Helper method to create an sms link

### PR DESCRIPTION
**Pull request by aantix**
When the user clicks the sms link, the user's messaging software is opened with the passed in phone number and optional body value prepopulated. 

Test page :
https://sms-to-rails-test.herokuapp.com/

```
<%= sms_to "15155555785", "Text me", body: "I have a question.." %>
```

### Summary

OSX Chrome/Safari
image

Android
image

iOS Safari
image

### CODE REVIEW

**Comment by guilleiguaran** [comment]
I think this should be `sms_to` to be consistent with `email_to`

**Comment by aantix** [comment]
> I think this should be `sms_to` to be consistent with `email_to`

Fair point.  Changed to `sms_to`.

**Comment by aantix** [comment]
Test page :
https://sms-to-rails-test.herokuapp.com/

**Review by sharang-d — COMMENTED** [review]
Simple doc changes :)

**Comment by aantix** [comment]
Are there any additional messaging clients that we can test this with?
https://sms-to-rails-test.herokuapp.com/

**Review by guilleiguaran — APPROVED** [review]

**Comment by pdobb** [comment]
🤜🏻🤛🏻

**Comment by bogdan** [comment]
`body` escaping had some problems in historical versions of iOS and Android.
We still have this workaround in our code:

``` ruby
    unless ios_greater_or_equal?("10", user_agent) ||   android_greater_or_equal?('6.0.1', user_agent) # problem was fixed in ios 10 and android 6.0.1+
      body.gsub!('%', '%25')
    end
```

Maybe this is still actual and should be supported.

File: actionview/lib/action_view/helpers/url_helper.rb [hunk 1]
```diff
@@ -571,6 +571,54 @@ def current_page?(options, check_parameters: false)
         end
       end
 
+      # Creates an SMS anchor link tag to the specified +phone_number+, which is
+      # also used as the name of the link unless +name+ is specified. Additional
+      # HTML attributes for the link can be passed in +html_options+.
+      #
+      # When clicked, an SMS message is prepopulated with the passed phone number
+      #  and optional +body+ value.
```
Comment (K1N5L4Y3R) [inline actionview/lib/action_view/helpers/url_helper.rb]: image

Watch out for indentation here (being parsed as a code block)

Comment (kamipo) [reply actionview/lib/action_view/helpers/url_helper.rb]: Good catch. Fixed by 0e149abbd5affc43f80a5c9bf8b7a7c2a5905d53.

**Comment by Nodalailama** [comment]
Hello,

There is a small problem: when I want to open sms app without any number, the ";" just before the "?" => "sms:;?" add in the phone field a blank phone number.

If you can remove the ";" when phone is blank then it work correctly.

Cordialement,

**Pull request merged**